v2/arangodb/database_impl.go

```diff
@@ -100,3 +99,24 @@ func (d database) Info(ctx context.Context) (DatabaseInfo, error) {
 		return DatabaseInfo{}, response.AsArangoErrorWithCode(code)
 	}
 }
+
+func (d database) TransactionJS(ctx context.Context, options TransactionJSOptions) (interface{}, error) {
+	url := d.url("_api", "transaction")
+
+	var transactionResponse struct {
+		shared.ResponseStruct `json:",inline"`
+		Result                interface{} `json:"result"`
+	}
+
+	resp, err := connection.CallPost(ctx, d.client.connection, url, &transactionResponse, &options)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	switch code := resp.Code(); code {
+	case http.StatusOK:
+		return transactionResponse.Result, nil
+	default:
+		return nil, transactionResponse.AsArangoError()
```

Comment:
FYI: alternative: `connection.CallWithChecks(ctx, d.client.connection, http.MethodPost, url, &transactionResponse, []int{http.StatusOK}, WithBody(&options))`.
Defining `shared.ResponseStruct` and switch would be unneeded in that case